src/main/java/com/questhelper/helpers/quests/dragonslayerii/DragonSlayerII.java

```diff
@@ -964,7 +964,7 @@ public void setupSteps()
 		talkToBobAfterDiningRoom = new NpcStep(this, NpcID.BOB_8111, new WorldPoint(3222, 3476, 0), "Talk to Bob in Varrock Castle.");
 
 		// Final battle
-		takeBoatToUngael = new NpcStep(this, NpcID.TORFINN_10405, new WorldPoint(2640, 3696, 0), "Talk to Torfinn on the Rellekka docks to go to Ungael. Be prepared for a lot of fighting.", combatGear, antifireShield);
+		takeBoatToUngael = new NpcStep(this, NpcID.TORFINN_10405, new WorldPoint(2640, 3696, 0), "Talk to Torfinn on the Rellekka docks to go to Ungael. Be prepared for a lot of fighting.", combatGear, antifireShield,  new ItemRequirement("Empty inventory spaces for tools to repair the ship", -1, 4).isNotConsumed());
```

Comment:
There's a `FreeInventorySlotRequirement` you can use instead of the `ItemRequirement`

Reply:
Fixed in latest commit. lmk if I did it wrong or should use a different style of syntax.